app/models/update_log.rb

```diff
@@ -4,14 +4,31 @@ def self.setting_record
     Setting.find_or_create_by(key: 'metrics_update')
   end
 
-  def self.log_update(time)
+  def self.last_update
+    setting = self.setting_record
+    setting.value.values.last
+
+  end
+
+  def self.log_updates(time)
     setting = self.setting_record
-    setting.value["last_update"] = time
+    last_update = setting.value.keys.last
+    if last_update === nil
+      new_update = 0
+    elsif last_update >= 10
```

Comment:
This number is a "magic number", which should be replaced by a constant, so that the name can help show what it's for — in this case, the number of updates to keep.